### ISSUE TITLE
required textfield without asterisk

### ISSUE THREAD

**Issue by VinceMalone** [material-components/material-components-web#1363]
The [MD spec mentions](https://material.io/guidelines/components/text-fields.html#text-fields-layout) forms that have a majority of required fields do not need to be visually indicated with an asterisk (*).
> In forms with some required fields, indicate all of the required ones. However, if there are fewer optional fields, indicate those instead.

### mdc-textfield

Currently if `.mdc-textfield__input` is `:required`, the asterisk will always appear after the component's label. There is no way to keep the input as `:required` and [not have an asterisk in the label](https://github.com/material-components/material-components-web/blob/v0.21.1/packages/mdc-textfield/mdc-textfield.scss#L461).

One way to fix this is to allow a modifier class to omit the asterisk when required:
```css
.mdc-textfield__input:not(.mdc-textfield__input--not-visually-required):required + .mdc-textfield__label::after { ... }
/* or */
.mdc-textfield__input:required + .mdc-textfield__label:not(.mdc-textfield__label--conceal-required)::after { ... }
```
Though it's probably worth discussing what the best way to implement is.

**Comment by tinybigideas** [comment]
Coming from: https://github.com/trimox/angular-mdc-web/issues/2216

There's an asterisk added by CSS even if there is no label. More details in the above issue.

**Comment by asyncLiz** [comment]
I don't believe this is the same problem. This issue is a design question and not a bug.

My guess for Angular would be that the label _element_ is still there even if the label itself is not present, which is why the CSS is able to add an asterisk to it. In MDC this isn't an issue because the label element is not added to the DOM unless a label is provided.

[This line](https://github.com/trimox/angular-mdc-web/blob/1d824b58f3ecf6f5f34f5b14da9030b900afdebb/packages/textfield/text-field.html#L31) would make me suspicious since the `ngIf` does not check the truthiness of the label value itself.

**Comment by tinybigideas** [comment]
@asyncLiz oh, thank you very much. I overlooked! Ignore my comment.
